crates/rnote-cli/src/export.rs

```diff
@@ -0,0 +1,715 @@
+        /// quality of the jpeg image
+        #[arg(long, default_value_t = 85)]
+        jpeg_quality: u8,
+    },
+    /// Export selection of a document {n}
+    /// When using --output-file, only one input file can be given.{n}
+    /// The export format is recognized from the file extension of the output file.{n}
+    /// When using --output-format, the same file name is used with the extension changed.{n}
+    /// --output-file and --output-format are mutually exclusive but one of them is required.{n}
+    /// Usages: {n}
+    /// rnote-cli export doc --output-file [filename.(svg|xopp|pdf)] [1 file] selection {n}
+    /// rnote-cli export doc --output-format [svg|xopp|pdf] [list of files] selection {n}
+    /// Available selection args: only use one of them {n}
+    /// --all: select all strokes {n}
+    /// --rect X Y deltaX deltaY : Select all strokes in given area {n}
+    Selection {
+        #[command(flatten)]
+        file: FileArgs<SelectionOutputFormat>,
+        #[command(flatten)]
+        selection: SelectionArgs,
+        #[arg(short = 'i', long, default_value_t = Bounds::default(), conflicts_with = "all")]
+        /// if the lines inside or intersecting with the given bounds are exported. Exclusive with --all
+        bounds: Bounds,
+        /// bitmap scale factor in relation to the actual size on the document
+        #[arg(long, default_value_t = 1.8)]
+        bitmap_scalefactor: f64,
+        /// quality of the jpeg image
+        #[arg(long, default_value_t = 85)]
+        jpeg_quality: u8,
+        /// margin around the document
+        #[arg(long, default_value_t = 12.0)]
+        margin: f64,
+    },
+}
+
+#[derive(Args, Debug)]
+#[group(required = true, multiple = false)]
+pub(crate) struct FileArgs<T: ValueEnum + 'static + Send + Sync> {
+    /// the export output file. Only allows for one input file. Exclusive with --output-format
+    #[arg(short = 'o', long)]
+    output_file: Option<PathBuf>,
+    /// the export output format. Exclusive with --output-file
+    #[arg(short = 'f', long)]
+    output_format: Option<T>,
+}
+
+#[derive(Args, Debug)]
+#[group(required = true, multiple = false)]
+pub(crate) struct SelectionArgs {
+    /// export all strokes. Exclusive with --rect
+    #[arg(short = 'a', long, action = ArgAction::SetTrue)]
+    all: bool,
+    /// Export an rectengular area of the canvas, exclusive with --all  {n}
+    /// usage: X Y deltaX deltaY {n}
+    /// Goes to given coordiates and selects all strokes in a given rectangle based of the given delta values
+    #[arg(short = 'r', long, value_name = "X", num_args = 4)]
+    rect: Option<Vec<f64>>,
+}
+
+#[derive(ValueEnum, Clone, Debug, Default)]
+pub(crate) enum Bounds {
+    #[default]
+    Inside,
+    Intersecting,
+}
+
+impl Display for Bounds {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", format!("{self:?}").to_lowercase())
+    }
+}
+
+#[derive(ValueEnum, Clone, Debug)]
```

Comment:
I tried that but seems like features don't work well with workspace dependency and I also couldn't get optional dependencies working directly

Reply:
Hm, I don't think it is that complicated. I quickly implemented it for `DocExportFormat`. Check out the `clap-derive-showcase` branch to see how it can work.

Reply:
see https://github.com/flxzt/rnote/commit/33315db8eaaad7bd737c296065a5e37673d537c6#commitcomment-123206327